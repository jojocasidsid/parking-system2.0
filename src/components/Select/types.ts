import { Control } from 'react-hook-form'

export interface IOptions {
	label: string
	value: string | number
}
export interface IProps {
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	control?: Control<any, object> // To do: change any type
	children: React.ReactNode
	defaultValue?: string
	disabled?: boolean
	error?: string
	fullWidth?: boolean
	label?: string
	name: string
	placeholder?: string
	required?: boolean
	noController?: boolean
	options: IOptions[]
}
