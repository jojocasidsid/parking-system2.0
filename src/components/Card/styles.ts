import styled from '@emotion/styled'

export const CardWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlotHeader = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`
