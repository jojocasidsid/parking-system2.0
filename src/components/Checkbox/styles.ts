import styled from '@emotion/styled'

export const StyledRoot = styled('div')`
	padding: 18px;
`
