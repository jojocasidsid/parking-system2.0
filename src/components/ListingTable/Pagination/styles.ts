import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export interface IPageInputProps {
	error?: string
}

export const PageInput = styled.input<IPageInputProps>`
	border-radius: 2px 0px 0px 2px;
	padding: 7px 2px 7px 0px;
	text-align: center;
	width: 100%;
	font-style: normal;
	font-size: 14px;
	line-height: 16px;
	font-weight: 500;
	font-stretch: normal;
	height: 28px;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&:active,
	&:focus {
		border: 1px solid ${({ theme }) => theme.palette.primary.main};
		outline: none;
		box-shadow: none;
	}

	&:disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	border: 1px solid
		${({ theme, error }) => {
			if (error) {
				return theme?.palette.error.main
			}
			return theme.palette.text.secondary
		}};
`

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
	font-size: 12px;
`
