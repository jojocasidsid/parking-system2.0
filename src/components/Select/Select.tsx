import React from 'react'
import { Controller } from 'react-hook-form'

import { MenuItem } from '@mui/material'
import { IProps } from './types'

import {
	StyledRoot,
	StyledLabel,
	StyledSelect,
	StyledArrowDown,
	StyledFormHelper,
} from './styles'

const Select = ({
	control,

	defaultValue,
	disabled,
	error,
	fullWidth = false,
	label,
	required = false,
	name,

	options,
}: IProps) => (
	<Controller
		name={name}
		control={control}
		defaultValue={defaultValue}
		render={({ field: { onChange, onBlur, value } }) => (
			<StyledRoot>
				{label && (
					<StyledLabel htmlFor={name}>
						{label} <span>{required && '*'}</span>
					</StyledLabel>
				)}
				<StyledSelect
					id={name}
					disabled={disabled}
					onChange={onChange}
					onBlur={onBlur}
					value={value === -1 ? '' : value} // Workaround for out-of-range warnings, initial values for select inputs of number types are -1 as empty strings are not assignable
					fullWidth={fullWidth}
					// eslint-disable-next-line react/no-unstable-nested-components
					IconComponent={(props) => (
						<StyledArrowDown
							src='/assets/arrow-down.svg'
							alt='arrow-down'
							{...props}
						/>
					)}
					error={Boolean(error)}
				>
					{options.map((row, id) => (
						<MenuItem key={id} value={row.value}>
							{row.label}
						</MenuItem>
					))}
				</StyledSelect>
				<StyledFormHelper error={Boolean(error)}>{error}</StyledFormHelper>
			</StyledRoot>
		)}
	/>
)

export default Select
