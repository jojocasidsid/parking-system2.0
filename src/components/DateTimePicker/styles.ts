import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface ITheme {}
interface IInput extends ITheme {
	errors?: boolean
	theme?: Theme
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, errors }: IInput) => {
		if (errors) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: 8px 16px 8px 16px;
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
		border-color: ${({ theme, errors }: IInput) =>
			errors
				? theme?.palette.error.main
				: theme?.palette.text.secondary} !important;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 8px;
	}

	.MuiFormHelperText-root {
		color: ${({ theme }) => theme.palette.error.main};
		margin-left: 0;
	}
`
