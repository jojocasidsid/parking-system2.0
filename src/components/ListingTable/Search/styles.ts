import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface IInput {
	error?: boolean
	theme?: Theme
	small?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, error }: IInput) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: ${({ small }: IInput) =>
			small ? '2px 8px 2px 8px' : '8px 12px 8px 16px'};
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme?.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px !important;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 0px;
	}

	.MuiInputBase-adornedEnd div.MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
		padding-right: 0px !important;
	}

	.MuiInputBase-adornedStart {
		padding-left: 0px !important;
	}

	.MuiInputBase-adornedStart .MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
	}

	.MuiFormHelperText-root {
		margin-left: 0;
	}
`
