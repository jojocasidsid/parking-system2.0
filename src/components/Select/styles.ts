import styled from '@emotion/styled'
import { Theme } from '@mui/material/styles'
import { Select, FormHelperText } from '@mui/material'

interface IStyled {
	theme?: Theme
}
interface IStyledSelect extends IStyled {
	error?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled('label')`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;
	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;

	//has errors
	color: ${({ theme, error }: IStyledSelect) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.secondary
	}};

	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledSelect = styled(Select)`
	min-width: 120px;

	& .MuiSelect-select {
		padding: 8px 16px 8px 16px;
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	&.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}
`

export const StyledArrowDown = styled('img')`
	width: 13px;
`

export const StyledFormHelper = styled(FormHelperText)`
	color: ${({ theme }) => theme.palette.error.main};
`
