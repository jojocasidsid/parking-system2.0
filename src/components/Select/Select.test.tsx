import React from 'react'

// packages
import { ThemeProvider } from '@mui/material/styles'
import { render, screen } from '@testing-library/react'
import { useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import * as yup from 'yup'
import renderer from 'react-test-renderer'

// config
import theme from 'config/theme'
import Select from './Select'

const ERROR_FIELD_REQUIRED = 'This field is required'
export const validationSchema = yup.object({
	test: yup.string().required(ERROR_FIELD_REQUIRED).nullable(),
	test1: yup.string().required(ERROR_FIELD_REQUIRED).nullable(),
})
export interface IValidationSchema
	extends yup.InferType<typeof validationSchema> {}
export const defaultValues: IValidationSchema = {
	test: '',
	test1: 'test1',
}

describe('Input component', () => {
	it('should render correctly with a name on it', () => {
		const Component = () => {
			const {
				control,

				formState: { errors },
			} = useForm({
				mode: 'onTouched',
				defaultValues,
				resolver: yupResolver(validationSchema),
			})

			return (
				<ThemeProvider theme={theme}>
					<Select
						control={control}
						error={errors.test?.message}
						fullWidth
						required
						label='Test'
						name='test'
						options={[
							{ label: 'SP', value: 1 },
							{ label: 'MP', value: 1 },
							{ label: 'LP', value: 1 },
						]}
					/>
				</ThemeProvider>
			)
		}

		render(<Component />)

		const input = screen.getByRole('button') as HTMLInputElement

		expect(input).toBeVisible()
	})

	it('should create snapshot correctly', () => {
		const Component = () => {
			const {
				control,

				formState: { errors },
			} = useForm({
				mode: 'onTouched',
				defaultValues,
				resolver: yupResolver(validationSchema),
			})

			return (
				<ThemeProvider theme={theme}>
					<Select
						control={control}
						error={errors.test?.message}
						fullWidth
						required
						label='Test'
						name='test'
						options={[
							{ label: 'SP', value: 1 },
							{ label: 'MP', value: 1 },
							{ label: 'LP', value: 1 },
						]}
					/>
				</ThemeProvider>
			)
		}

		const tree = renderer.create(<Component />)

		expect(tree).toMatchSnapshot()
	})
})
