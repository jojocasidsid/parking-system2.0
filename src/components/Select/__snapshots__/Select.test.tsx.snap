// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should create snapshot correctly 1`] = `
<div
  className="css-evn1e9"
>
  <label
    className="css-1nafofr"
    htmlFor="test"
  >
    Test
     
    <span>
      *
    </span>
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-q9ksy9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-jcfq8n-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="test"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
  <p
    className="MuiFormHelperText-root css-11e8ze7-MuiFormHelperText-root"
  />
</div>
`;
